### ISSUE TITLE
Configuring reek for ruby version

### ISSUE THREAD

**Issue by klobuczek** [troessner/reek#1325]
How do you configure reek for a specific ruby version? I am using jruby-9.2.0.0 which is ruby 2.5 compatible and I am getting 
```
This file has unexpected token kRESCUE [reek]
```
on something like 
```
...each.do 
rescue ...
end
```

**Comment by Drenmi** [comment]
I'm guessing that period in between `each` and `do` is there in the non-redacted code as well, as running Reek on:

```ruby
foo each.do 
  rescue "bar"
end
```

gives the exact error you posted. 🙂

**Comment by troessner** [comment]
@klobuczek our codeclimate engine seems indeed to be broken unfortunately in some contexts. We will fix this right after Reek 5 [will be released](https://github.com/troessner/reek/issues/1305) which is expect to happen in the next couple of days since there are only 2 pull requests left to deal with.

**Comment by mvz** [comment]
@klobuczek are you experiencing this problem on the CodeClimate site or are you running the CodeClimate enginge locally?

**Comment by mvz** [comment]
@maxjacobson can you tell us which version of Reek CodeClimate is using? I think this problem is fixed already in Reek 4.8.1.

**Comment by maxjacobson** [comment]
:wave: looks like we're locked down to https://github.com/troessner/reek/commit/6a602abedff1075dc2328b7eac36800018d437c0, which is not quite up-to-date  (it seems to be a few commits shy of 4.7.1). We can update reek to the newer version, but we'll need to do a little QA before rolling it out. I can put that on our roadmap and try to make it happen soon

**Comment by mvz** [comment]
That would be great, @maxjacobson. How can we ensure CodeClimate stays up-to-date in the future? And should we just direct people with CodeClimate related issues to CodeClimate support?

**Comment by maxjacobson** [comment]
> How can we ensure CodeClimate stays up-to-date in the future? 

Any of these would work:

- let us know of the new release via our support form (https://codeclimate.com/help)
- let us know of the new release via our community slack (you can join by filling out this form: https://codeclimate.com/dev_form/ and we can set up a reek-specific channel)
- let us know of the new release by opening an issue on https://github.com/codeclimate/codeclimate

> And should we just direct people with CodeClimate related issues to CodeClimate support?

That would be totally fine, and we'd definitely appreciate being made aware of any issues folks are having. Directing folks to https://codeclimate.com/help would be best.

**Comment by mvz** [comment]
Thanks, @maxjacobson.

**Comment by troessner** [comment]
AFAIS can see the current Reek CC integration is broken. Our own pull requests #1319 fail with errors like in the attached screenshot.
<img width="961" alt="screen shot 2018-06-20 at 21 32 56" src="https://user-images.githubusercontent.com/271109/41680425-c0a09434-74d1-11e8-8b0a-56bfdd62c1a8.png">

**Comment by troessner** [comment]
Best would be to wait for Reek 5 though, then we start with a clean slate @maxjacobson

**Comment by mvz** [comment]
No, our own integration is a special case because of the changed contents of `defaults.reek`.

**Comment by mvz** [comment]
@troessner  You can see that by running released Reek 4.8.1 on master:

```
$ reek lib
You are trying to configure smell type enabled but we can't find one with that name.
Please make sure you spelled it right. (See 'defaults.reek' in the Reek
repository for a list of all available smell types.)
```

**Comment by maxjacobson** [comment]
FWIW, the 4.8.1 release went out today. The QA looked good with a modest amount of testing. If Reek 5 has breaking changes for folks with existing checked-in config, we may want to release it as a special channel, as we do with the rubocop engine :thinking:

**Comment by troessner** [comment]
@klobuczek did this upgrade solve your problem?

**Comment by klobuczek** [comment]
@troessner yes, it is 2.5 compatible now, sorry for the late response
